artemis-commons/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQScheduledComponent.java

```diff
@@ -188,41 +202,53 @@ public ClassLoader run() {
 
    }
 
+   /**
+    * If {@code overBooking} is {@code true} a delay request can succeed only if:
+    * <ul>
+    *    <li>there is no other pending delay request
+    *    <li>there is no pending execution request
+    * </ul>
+    * <p>
+    * If {@code overBooking} is {@code false} a delay request must always succeed.<br>
+    * When a delay request succeed it schedule a new execution to happen in {@link #getPeriod()}.<br>
+    */
    public void delay() {
-      int value = delayed.incrementAndGet();
-      if (value > 10) {
-         delayed.decrementAndGet();
-      } else {
-         // We only schedule up to 10 periods upfront.
-         // this is to avoid a window where a current one would be running and a next one is coming.
-         // in theory just 2 would be enough. I'm using 10 as a precaution here.
-         scheduledExecutorService.schedule(runForScheduler, Math.min(period, period * value), timeUnit);
+      final AtomicBoolean bookedForRunning = this.bookedForRunning;
+      if (bookedForRunning == null || bookedForRunning.compareAndSet(false, true)) {
+         try {
+            scheduledExecutorService.schedule(() -> delayedRunForScheduler(bookedForRunning), nsPeriod, TimeUnit.NANOSECONDS);
+         } catch (RejectedExecutionException e) {
+            if (bookedForRunning == null) {
```

Comment:
Guessing that was meant to be != null? EDIT: as later commenting, think the null check should be eliminated through simplification.

Reply:
And indeed it was, good eyes!